hatchways/team-dumplings
# Implement test 🧪 for `logout` controller #1

### What this PR does :
-  mocha test script for `logout` controller
-  configure package.json to load the test script
- update  `README.md`, to describe how to run the tests

### Test output:
```
  /POST /auth/logout/
GET /auth/logout/ 200 3.966 ms - 32
    √ it should return 200  and  message


  1 passing (26ms)

MongoDB Connected: cluster0-shard-00-00.w4epm.mongodb.net
[nodemon] restarting due to changes...
[nodemon] starting `mocha ./test -r dotenv/config`
```

### information needed to test this feature :
-  Go into the server directory `cd server`
-  Run `npm run test` to start tests
```
npm run test
```

### issues with the current functionality :
### The need to perform more test scenarios 
 -  Verify all UI components are visible properly after logout
 - Check if the logs contain a denial of service attack for the logout

### Features:
- [ ] Prevent unauthorized IP from making requests for the logout  🔍  
- [ ] Update  needed to the frontend API calls for the logout route > `get` to `post` ⬆️
